src/components/VDisk/utils.ts

```diff
@@ -1,30 +1,24 @@
-import {getDefaultNodePath} from '../../containers/Node/NodePages';
 import {getVDiskPagePath} from '../../routes';
-import type {TVDiskStateInfo, TVSlotId} from '../../types/api/vdisk';
 import {valueIsDefined} from '../../utils';
-import {stringifyVdiskId} from '../../utils/dataFormatters/dataFormatters';
-import {isFullVDiskData} from '../../utils/disks/helpers';
+import type {PreparedVDisk} from '../../utils/disks/types';
 
-export function getVDiskLink(data: TVDiskStateInfo | TVSlotId) {
+export function getVDiskLink(data: PreparedVDisk) {
     let vDiskPath: string | undefined;
 
-    const isFullData = isFullVDiskData(data);
-    const VDiskSlotId = isFullData ? data.VDiskSlotId : data.VSlotId;
-
     if (
-        valueIsDefined(VDiskSlotId) &&
+        valueIsDefined(data.VDiskSlotId) &&
         valueIsDefined(data.PDiskId) &&
         valueIsDefined(data.NodeId)
     ) {
-        vDiskPath = getVDiskPagePath(VDiskSlotId, data.PDiskId, data.NodeId);
-    } else if (valueIsDefined(data.NodeId) && isFullVDiskData(data)) {
-        vDiskPath = getDefaultNodePath(
-            data.NodeId,
+        vDiskPath = getVDiskPagePath(data.VDiskSlotId, data.PDiskId, data.NodeId);
+    } else if (valueIsDefined(data.StringifiedId)) {
+        vDiskPath = getVDiskPagePath(
+            undefined as unknown as string,
+            data.PDiskId as number,
+            data.NodeId as number,
```

Comment:
You can instead modify `getVDiskPagePath` function and pass params without type assertions
```
export function getVDiskPagePath(
    params:
        | {
              vDiskSlotId: string | number;
              pDiskId: string | number;
              nodeId: string | number;
          }
        | {pDiskId?: string | number; nodeId?: string | number; vDiskId: string | number},
    query: Query = {},
) {
    return createHref(routes.vDisk, undefined, {...query, ...params});
}
```